modules/apps/layout/layout-reports-web/jest-setup.config.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+window.Liferay.Util.sub = function (string = '', data) {
```

Comment:
So we have similar logic now in:

- [`modules/dxp/apps/segments/segments-experiment-web/jest-setup.config.js`](https://github.com/liferay/liferay-portal/blob/ac84f4eb274124c758e8b3f029efc49b00778707/modules/dxp/apps/segments/segments-experiment-web/jest-setup.config.js)
- [`modules/dxp/apps/analytics/analytics-reports-web/jest-setup.config.js`](https://github.com/liferay/liferay-portal/blob/ac84f4eb274124c758e8b3f029efc49b00778707/modules/dxp/apps/analytics/analytics-reports-web/jest-setup.config.js)

I'm making an issue to remove this duplication by baking a reasonable substitute into our default mock environment. Will come back and update the link here once I've done that.

Reply:
Here's the link: https://github.com/liferay/liferay-frontend-projects/issues/519

Reply:
That config is used to substitute the `x` in the message keys to check the full text in the tests. It will be awesome to not be required to add it each module. I saw that we are the only team using it in our tests 🤔 `layout-reports-web`, `segments-experiment-web` and `analytics-reports-web`.